vm/x86/x86emu/src/registers.rs

```diff
@@ -75,127 +74,28 @@ impl CpuState {
-        } else if reg < Register::AH {
-            self.gps[reg.number()] as u8
-        } else {
-            (self.gps[reg.number() - 4] >> 8) as u8
-        }
-    }
-
-    fn set_gp64(&mut self, reg: Register, val: u64) {
-        debug_assert!(reg.is_gpr64());
-        self.gps[reg.number()] = val;
-    }
-
-    fn set_gp32(&mut self, reg: Register, val: u32) {
-        debug_assert!(reg.is_gpr32());
-        // N.B. setting a 32-bit register zero extends the result to the 64-bit
-        //      register. This is different from 16-bit and 8-bit registers.
-        self.gps[reg.number()] = val as u64;
-    }
-
-    fn set_gp16(&mut self, reg: Register, val: u16) {
-        debug_assert!(reg.is_gpr16());
-        self.gps[reg.number()] &= !0xffff;
-        self.gps[reg.number()] |= val as u64;
-    }
-
-    fn set_gp8(&mut self, reg: Register, val: u8) {
-        debug_assert!(reg.is_gpr8());
-        if reg >= Register::SPL {
-            self.gps[reg.number() - 4] &= !0xff;
-            self.gps[reg.number() - 4] |= val as u64;
-        } else if reg < Register::AH {
-            self.gps[reg.number()] &= !0xff;
-            self.gps[reg.number()] |= val as u64;
-        } else {
-            self.gps[reg.number() - 4] &= !0xff00;
-            self.gps[reg.number() - 4] |= (val as u64) << 8;
-        }
-    }
-
-    pub(crate) fn get_gp(&self, reg: Register) -> u64 {
-        if reg.is_gpr64() {
-            self.get_gp64(reg)
-        } else if reg.is_gpr32() {
-            self.get_gp32(reg).into()
-        } else if reg.is_gpr16() {
-            self.get_gp16(reg).into()
-        } else {
-            debug_assert!(reg.is_gpr8());
-            self.get_gp8(reg).into()
-        }
-    }
-
-    pub(crate) fn get_gp_sign_extend(&self, reg: Register) -> i64 {
-        if reg.is_gpr64() {
-            self.get_gp64(reg) as i64
-        } else if reg.is_gpr32() {
-            (self.get_gp32(reg) as i32).into()
-        } else if reg.is_gpr16() {
-            (self.get_gp16(reg) as i16).into()
-        } else {
-            debug_assert!(reg.is_gpr8());
-            (self.get_gp8(reg) as i8).into()
-        }
-    }
-
-    pub(crate) fn set_gp(&mut self, reg: Register, val: u64) {
-        if reg.is_gpr64() {
-            self.set_gp64(reg, val)
-        } else if reg.is_gpr32() {
-            self.set_gp32(reg, val as u32)
-        } else if reg.is_gpr16() {
-            self.set_gp16(reg, val as u16)
-        } else {
-            debug_assert!(reg.is_gpr8());
-            self.set_gp8(reg, val as u8)
-        }
-    }
-
-    pub(crate) fn bitness(&self) -> Bitness {
-        if self.cr0 & x86defs::X64_CR0_PE != 0 {
-            if self.efer & x86defs::X64_EFER_LMA != 0 {
-                if self.segs[CpuState::CS].attributes.long() {
-                    Bitness::Bit64
-                } else {
-                    Bitness::Bit32
-                }
+pub fn bitness(cr0: u64, efer: u64, cs: SegmentRegister) -> Bitness {
```

Comment:
Does this need to be pub or can it stay pub(crate)?

Reply:
Fixed